test/sequential/xyk-pallet.api.maintenance-rules.test.ts

```diff
@@ -0,0 +1,162 @@
+/*
+ *
+ * @group maintenance
+ */
+import { getApi, initApi } from "../../utils/api";
+import {
+  ExtrinsicResult,
+  expectMGAExtrinsicSuDidSuccess,
+} from "../../utils/eventListeners";
+import { BN } from "@polkadot/util";
+import { User } from "../../utils/User";
+import {
+  setupApi,
+  setup5PoolsChained,
+  Extrinsic,
+  setupUsers,
+} from "../../utils/setup";
+import { getEventResultFromMangataTx } from "../../utils/txHandler";
+import { BN_ONE, BN_HUNDRED, signTx } from "@mangata-finance/sdk";
+import { FOUNDATION_ADDRESS_1, MGA_ASSET_ID } from "../../utils/Constants";
+import { BN_MILLION } from "@mangata-finance/sdk";
+import { Sudo } from "../../utils/sudo";
+import { Xyk } from "../../utils/xyk";
+import { ApiPromise } from "@polkadot/api";
+import { Assets } from "../../utils/Assets";
+import {
+  AggregatorOptions,
+  Staking,
+  tokenOriginEnum,
+} from "../../utils/Staking";
+import { Maintenance } from "../../utils/Maintenance";
+import { getLiquidityAssetId } from "../../utils/tx";
+
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.setTimeout(1500000);
+
+let users: User[] = [];
+let tokenIds: BN[] = [];
+let api: ApiPromise;
+let swapOperations: { [K: string]: Extrinsic } = {};
+let testUser1: User;
+let testUser2: User;
+let minStk: BN;
+const foundationAccountAddress = FOUNDATION_ADDRESS_1;
+describe("On Maintenance mode - multiSwaps / swaps / compound / prov liq are not allowed", () => {
+  beforeAll(async () => {
+    try {
+      getApi();
+    } catch (e) {
+      await initApi();
+    }
+    await setupApi();
+    ({ users, tokenIds } = await setup5PoolsChained(users));
+    api = await getApi();
+    const liq = await getLiquidityAssetId(tokenIds.slice(-1)[0], MGA_ASSET_ID);
+    swapOperations = {
+      multiswapSellAsset: Xyk.multiswapSellAsset(tokenIds, BN_HUNDRED, BN_ONE),
+      multiswapBuyAsset: Xyk.multiswapBuyAsset(
+        tokenIds,
+        BN_HUNDRED,
+        BN_MILLION
+      ),
+      sellAsset: Xyk.sellAsset(tokenIds[0], tokenIds[1], BN_HUNDRED, BN_ONE),
+      buyAsset: Xyk.buyAsset(tokenIds[0], tokenIds[1], BN_HUNDRED, BN_MILLION),
+      //      compoundRewards: Xyk.compoundRewards(liq),
+      provideLiquidity: Xyk.provideLiquidity(liq, MGA_ASSET_ID, BN_HUNDRED),
+    };
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(
+        foundationAccountAddress,
+        Maintenance.switchMaintenanceModeOn()
+      ),
+      Xyk.updatePoolPromotion(liq, 20)
+    ).then((value) => {
+      expectMGAExtrinsicSuDidSuccess(value);
+    });
+  });
+  it.each([
+    "multiswapSellAsset",
```

Comment:
An error `"1010: Invalid Transaction: The swap prevalidation has failed"` occurs when trying to pass the test by commands:  "multiswapSellAsset",  "multiswapBuyAsset",  "compoundRewards", "provideLiquidity",

Reply:
Adapted. Not happy about the fix, but yes, some extrinsics fails on client, some others on tx execution. 
Hopefully this will be removed in the future.